File: Laboratorio03/python/041cbdCRUD.py
import os
import time
import mysql.connector

# Conexión a la base de datos
mydb = mysql.connector.connect(
    host="localhost",
    user="root",
    password="",
    database="sistemas"
)
mycursor = mydb.cursor()

while True:
    print('--MENU DEL SISTEMA---\n'
          ' 1. Insertar un dato.\n'
          ' 2. Eliminar un dato.\n'
          ' 3. Buscar un dato.\n'
          ' 4. Sobreescribir un dato.\n'
          ' 5. Mostrar el contenido de la lista.\n'
          ' 6. Salir del sistema.')
    
    opcion = int(input('Elige una opción: '))
    
    if opcion == 1:  # Create (insertar)
        dni = input('Ingresa el dni a insertar: ')
        nombres = input('Ingresa el nombre: ')
        apellidopa = input('Ingresa el apellido paterno: ')
        apellidoma = input('Ingresa el apellido materno: ')
        sql = "INSERT INTO personas (dni, nombres, apellidopa, apellidoma) VALUES (%s, %s, %s, %s)"
        val = (dni, nombres, apellidopa, apellidoma,)
        mycursor.execute(sql, val)
        mydb.commit()
        print(mycursor.rowcount, "registro insertado.")
        time.sleep(2)
        os.system('cls')
    
    elif opcion == 2:  # Delete (eliminar)
        dni = input('Ingresa el dni a eliminar: ')
        sql = "DELETE FROM personas WHERE dni = %s"
        val = (dni,)
        mycursor.execute(sql, val)
        mydb.commit()
        print(mycursor.rowcount, "registro(s) eliminado(s).")
        time.sleep(2)
        os.system('cls')
    
    elif opcion == 3:  # Read (leer)
        dni = input('Ingresa el dni a buscar: ')
        sql = "SELECT * FROM personas WHERE dni = %s"
        val = (dni,)
        mycursor.execute(sql, val)
        myresult = mycursor.fetchall()
        if myresult:
            print('El dni está en la tabla.')
        else:
            print('El dni no existe en la tabla.')
        time.sleep(2)
        os.system('cls')
    
    elif opcion == 4:  # UpDate (actualizar)
        datoAnterior = input('Ingresa el dni a sobreescribir: ')
        datoNuevo = input('Ingresa el nuevo dni: ')
        sql = "UPDATE personas SET dni = %s WHERE dni = %s"
        val = (datoNuevo, datoAnterior)
        mycursor.execute(sql, val)
        mydb.commit()
        print(mycursor.rowcount, "registro(s) actualizado(s).")
        time.sleep(2)
        os.system('cls')
    
    elif opcion == 5:  # Read (leer)
        mycursor.execute("SELECT * FROM personas")
        myresult = mycursor.fetchall()
        print("Contenido de la tabla:")
        for x in myresult:
            print(x)
        time.sleep(3)
        os.system('cls')
    
    elif opcion == 6:  # Salir
        respuesta = input('Estás seguro? (s/n)')
        if respuesta.upper() == 'S':
            print('Saliendo del sistema...')
            time.sleep(2)
            os.system("cls")
            break
        time.sleep(2)
        os.system('cls')
    else:
        print('Opción no válida, intenta de nuevo...')
        time.sleep(2)
        os.system('cls')

# Cerrar la conexión
mydb.close()

File: cpp/041cbdCRUD.cpp
// CRUD (create, Read, Update, Delete) con conexion a Base de datos
// Libreria a utlizar: pip mysql-connector-python

/*
    CREATE DATABASE sisteams;
    USE sistemas;
    CREATE TABLE personas (
        dni varchar(8) NOT NULL,
        apellidopa varchar(50) NOT NULL,
        apellidoma varchar(50) NOT NULL
    ) ENGINE = InnoDB DEFAULT CHARSET = utfmb4 COLLATE = utf8mb4_general_ci;

    ALTER TABLE personas
        ADD PRIMARY KEY (dni);
    COMMIT;
*/


#include <iostream>
#include <string>
#include <windows.h>

using namespace std;

void insertarDato() {
    string dni, nombres, apellidopa, apellidoma;

    cout << "Ingresa el dni a insertar: ";
    cin >> dni;
    cout << "Ingresa el nombre: ";
    cin >> nombres;
    cout << "Ingresa el apellido paterno: ";
    cin >> apellidopa;
    cout << "Ingresa el apellido materno: ";
    cin >> apellidoma;

    string query = "INSERT INTO personas (dni, nombres, apellidopa, apellidoma) VALUES ('" + dni + "', '" + nombres + "', '" + apellidopa + "', '" + apellidoma + "')";
    system(("C:\\xampp\\mysql\\bin\\mysql -u root -p sistemas -e \"" + query + "\"").c_str());
}

void eliminarDato() {
    string dni;

    cout << "Ingresa el dni a eliminar: ";
    cin >> dni;

    string query = "DELETE FROM personas WHERE dni='" + dni + "'";
    system(("C:\\xampp\\mysql\\bin\\mysql -u root -p sistemas -e \"" + query + "\"").c_str());
}

void buscarDato() {
    string dni;

    cout << "Ingresa el dni a buscar: ";
    cin >> dni;

    string query = "SELECT * FROM personas WHERE dni='" + dni + "'";
    int result = system(("C:\\xampp\\mysql\\bin\\mysql -u root -p sistemas -e \"" + query + "\"").c_str());
    if (result == 0) {
        cout << "El dni está en la tabla." << endl;
    } else {
        cout << "El dni no existe en la tabla." << endl;
    }
}

void sobreescribirDato() {
    string datoAnterior, datoNuevo;

    cout << "Ingresa el dni a sobreescribir: ";
    cin >> datoAnterior;
    cout << "Ingresa el nuevo dni: ";
    cin >> datoNuevo;

    string query = "UPDATE personas SET dni='" + datoNuevo + "' WHERE dni='" + datoAnterior + "'";
    system(("C:\\xampp\\mysql\\bin\\mysql -u root -p sistemas -e \"" + query + "\"").c_str());
}

void mostrarContenido() {
    system("C:\\xampp\\mysql\\bin\\mysql -u root -p sistemas -e \"SELECT * FROM personas\"");
}

int main() {
    int opcion;

    while (true) {
        cout << "--MENU DEL SISTEMA---\n"
                " 1. Insertar un dato.\n"
                " 2. Eliminar un dato.\n"
                " 3. Buscar un dato.\n"
                " 4. Sobreescribir un dato.\n"
                " 5. Mostrar el contenido de la lista.\n"
                " 6. Salir del sistema.\n";

        cout << "Elige una opción: ";
        cin >> opcion;

        switch (opcion) {
            case 1:
                insertarDato();
                break;
            case 2:
                eliminarDato();
                break;
            case 3:
                buscarDato();
                break;
            case 4:
                sobreescribirDato();
                break;
            case 5:
                mostrarContenido();
                break;
            case 6:
                char respuesta;
                cout << "Estás seguro? (s/n)";
                cin >> respuesta;
                if (toupper(respuesta) == 'S') {
                    cout << "Saliendo del sistema..." << endl;
                    Sleep(2000);
                    system("cls");
                    return 0;
                }
                Sleep(2000);
                system("cls");
                break;
            default:
                cout << "Opción no válida, intenta de nuevo..." << endl;
                Sleep(2000);
                system("cls");
                break;
        }
    }

    return 0;
}